package/MDAnalysis/guesser/base.py

```diff
@@ -0,0 +1,116 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding: utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from .. import _GUESSERS
+
+class GuesserMeta(type):
+    def __init__(cls, name, bases, classdict):
+        type.__init__(type, name, bases, classdict)
+        try:
+            _GUESSERS[classdict['context'].upper()] = cls
+        except KeyError:
+            pass
+
+class GuesserBase(metaclass=GuesserMeta):
+    context = 'base' 
+    _guess = {}
+
+    # give a rank to each atrribute based on its dependency on other attributes
+    # to be guessed, so that the attribute with lesser dependcy will be guessed
+    # first
+    _rank = {}
+
+    def __init__(self, atoms, **kwargs):
+        self._atoms = atoms
+        self._kwargs = kwargs
+
+    def is_guessed(self, guess):
```

Comment:
I do not like the name. I initially understood it as figuring out if the attribute was guessed or read from an input file. Something like `is_guessable` may be less ambiguous.